src/components/ColumnItem.tsx

```diff
@@ -0,0 +1,113 @@
+import { useState } from 'react'
+import { useFirestore } from 'react-redux-firebase'
+import { useSelector } from 'react-redux'
+import {
+  Paper,
+  IconButton,
+  TextField,
+  Typography,
+  Divider,
+} from '@mui/material'
+import CheckIcon from '@mui/icons-material/Check'
+import DeleteIcon from '@mui/icons-material/Delete'
+import EditIcon from '@mui/icons-material/Edit'
+import { CardList } from './CardList'
+import { CardForm } from './CardForm'
+import { Card, WithID, RootState } from '../store/store'
+
+export const ColumnItem = (props: {
+  cards: WithID<Card>[]
+  title: string
+  id: string
+  key: string
+}) => {
+  const firestore = useFirestore()
+  const { uid } = useSelector((state: RootState) => state.firebase.auth)
+
+  const columnCards = props.cards.filter((card) =>
+    card ? card.columnID === props.id : false
+  )
+
+  const [editMode, setEditMode] = useState(false)
+  const [presetColumnTitle, setPresetColumnTitle] = useState(props.title)
+
+  const handleChange = ({
+    currentTarget: { name, value },
+  }: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => {
+    if (name === 'editColumn') {
+      setPresetColumnTitle(value)
+    }
+  }
+
+  const editColumn = (columnTitle: string) => {
+    firestore
+      .collection('users')
+      .doc(uid)
+      .collection('columns')
+      .doc(props.id)
+      .update({ title: columnTitle })
+
+    setEditMode(false)
+  }
+
+  const deleteColumn = () => {
+    firestore
+      .collection('users')
+      .doc(uid)
+      .collection('columns')
+      .doc(props.id)
+      .delete()
+  }
+
+  return (
+    <Paper
+      sx={{
+        width: 330,
+        padding: '10px',
+        marginRight: '10px',
+        background: '#ebecf0',
+      }}
+    >
+      {editMode ? (
+        <form style={{ display: 'flex', justifyContent: 'space-between' }}>
```

Comment:
I'm wondering if you need that `form` tag here? Maybe it's not bad in terms of semantic HTML. But from the other hand, it would be great to omit inline-styles.

I think that you can use MUI `Box` component which could have prop `component="form"` like in [this example](https://mui.com/components/text-fields/#validation) (click `Show the source` in `Validation` example to see the code).